client/state/http/test/index.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import { HTTP_REQUEST } from 'state/action-types';
+import useNock, { nock } from 'test/helpers/use-nock';
+import { extendAction } from 'state/utils';
+import { failureMeta, successMeta } from 'state/data-layer/wpcom-http';
+import actionHandlers from '../';
+
+const httpHandler = actionHandlers[ HTTP_REQUEST ][ 0 ];
+
+const succeeder = { type: 'SUCCESS' };
+const failer = { type: 'FAIL' };
+
+const requestDomain = 'https://api.some-domain.com:443';
+const requestPath = '/endpoint';
+const getMe = {
+	url: requestDomain + requestPath,
+	method: 'GET',
+	onFailure: failer,
+	onSuccess: succeeder,
+};
+
+describe( '#httpHandler', () => {
+	let next;
+
+	useNock();
+
+	beforeEach( () => {
+		next = sinon.spy();
+	} );
+
+	it( 'should call `onSuccess` when a response returns with data', done => {
+		const data = { value: 1 };
+		nock( requestDomain ).get( requestPath ).reply( 200, data );
+
+		const dispatch = sinon.spy( () => {
+			expect( dispatch ).to.have.been.calledOnce;
+
+			sinon.assert.calledWith(
+				dispatch,
+				sinon.match(
+					extendAction(
+						succeeder,
+						successMeta( { body: data } )
+					)
+				)
+			);
+
+			done();
+		} );
```

Comment:
this is some strange use of the spy here. was there a specific reason to embed all the logic inside the spy instead of following-up after the handler call?

```js
const dispatch = spy();

httpHandler( { dispatch }, getMe, next );
expect( dispatch ).to.have.been.calledOnce;
expect( dispatch ).to.have.been.calledWithMatch( extendAction( succeeder, successMeta( { body: data } ) ) );
```

here we don't even need `done()`, which is usually reserved for asynchronous operations